version: '3.8'

services:
  db:
    container_name: db_task_management
    image: postgres:14.1-alpine
    restart: always
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
    ports:
      - '5432:5432'
    networks:
      - postgress_task_management
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${POSTGRES_USER}", "-d", "task_manager"]
      interval: 10s
      retries: 5
    volumes:
      - task_managment_db_data:/var/lib/postgresql/data

  backend:
    container_name: task_management_service
    depends_on:
      db:
        condition: service_healthy  
    build: .
    command: ./main  
    volumes:
       - ./src:/app/src
    ports:
      - '8000:8000'  
    env_file:
      - .env  
    networks:
      - task_management_service
networks:
  postgress_task_management:
    external: true
    name: task_management_name
  task_management_service:
    external: true
    name: ftask_management_name

volumes:
  task_managment_db_data: